external/solr/src/main/java/com/digitalpebble/storm/crawler/solr/persistence/SolrSpout.java

```diff
@@ -208,20 +218,19 @@ private void populateBuffer() {
 
                 Metadata metadata = new Metadata();
 
-                String mdAsString = (String) doc.get("metadata");
-                // get the serialized metadata information
-                if (mdAsString != null) {
-                    // parse the string and generate the MD accordingly
-                    // url.path: http://www.lemonde.fr/
-                    // depth: 1
-                    String[] kvs = mdAsString.split("\n");
-                    for (String pair : kvs) {
-                        String[] kv = pair.split(": ");
-                        if (kv.length != 2) {
-                            LOG.info("Invalid key value pair {}", pair);
-                            continue;
+                Iterator<String> keyIterators = doc.getFieldNames().iterator();
+                while (keyIterators.hasNext()) {
+                    String key = keyIterators.next();
+
+                    if (key.startsWith(prefix)) {
+                        Collection<Object> values = doc.getFieldValues(key);
+
+                        Iterator<Object> valueIterator = values.iterator();
+                        while (valueIterator.hasNext()) {
+                            String value = (String) valueIterator.next();
+
+                            metadata.addValue(StringUtils.replace(key, prefix, "", 1), value);
```

Comment:
the replacement could be done just once outside the while loop as it will be the same for all the values